dotcom-rendering/src/web/server/blocksToHtml.tsx

```diff
@@ -38,28 +41,37 @@ export const blocksToHtml = ({
 		adUnit,
 	});
 
+	const key = 'dcr';
+	const cache = createCache({ key });
+	const emotionServer = createEmotionServer(cache);
+
 	const html = renderToString(
-		<LiveBlogRenderer
-			blocks={blocks}
-			format={format}
-			adTargeting={adTargeting}
-			host={host}
-			pageId={pageId}
-			webTitle={webTitle}
-			ajaxUrl={ajaxUrl}
-			isSensitive={isSensitive}
-			isAdFreeUser={isAdFreeUser}
-			switches={switches}
-			isLiveUpdate={true}
-			section={section}
-			// The props below are never used because isLiveUpdate is true but, typescript...
-			shouldHideReaderRevenue={false}
-			tags={[]}
-			isPaidContent={false}
-			contributionsServiceUrl=""
-			keywordIds={keywordIds}
-		/>,
+		<CacheProvider value={cache}>
+			<LiveBlogRenderer
+				blocks={blocks}
+				format={format}
+				adTargeting={adTargeting}
+				host={host}
+				pageId={pageId}
+				webTitle={webTitle}
+				ajaxUrl={ajaxUrl}
+				isSensitive={isSensitive}
+				isAdFreeUser={isAdFreeUser}
+				switches={switches}
+				isLiveUpdate={true}
+				section={section}
+				// The props below are never used because isLiveUpdate is true but, typescript...
+				shouldHideReaderRevenue={false}
+				tags={[]}
+				isPaidContent={false}
+				contributionsServiceUrl=""
+				keywordIds={keywordIds}
+			/>
+		</CacheProvider>,
 	);
 
-	return html;
+	const chunks = emotionServer.extractCriticalToChunks(html);
+	const extractedCss = emotionServer.constructStyleTagsFromChunks(chunks);
+
+	return extractedCss + html;
```

Comment:
This just looks odd but I assume concatenating the CSS styles and HTML like this is fine as they are live blocks that get injected into the page?

Reply:
Yeah it seems to work! Perhaps using a template string, `return ${extractedCss}${html}` would read a bit better?